umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_smap/instrument.clj

```diff
@@ -0,0 +1,117 @@
+            instrument (xml-elem->child-instrument instrument-elem)]
+        (when (some #(= mountedOn-id-wo-# %) all-possible-instrument-ids)
+          [(str id "-" mountedOn-id-wo-#) instrument])))))
+
+(defn- xml-elem->instrument-sensor-mapping
+  [doc instrument-elem]
+  (let [all-possible-instrument-ids (keep #(get-in % [:attrs :id]) (select doc instruments-xpath1))
+        id (get-in instrument-elem [:attrs :id])
+        mountedOn-id (first (keep #(get-in % [:attrs :xlink/href]) (select instrument-elem "eos:mountedOn")))]
+    ;; only include child instruments - when mountedOn-id = one of the instrument ids.
+    (when mountedOn-id
+      (let [mountedOn-id-wo-# (.replaceAll mountedOn-id "#" "")
+            instrument (xml-elem->child-instrument instrument-elem)]
+        (when (some #(= mountedOn-id-wo-# %) all-possible-instrument-ids)
+          [(str id "-" mountedOn-id-wo-#) instrument])))))
+
+ (defn- xml-elem->child-instruments-mapping
+   "Returns the child instrument id-mountedOnid to child Instrument mapping 
+    by parsing the given xml element"
+  [doc]
+  (merge
+    (into {}
+          (map (partial xml-elem->child-instrument-mapping doc) (select doc instruments-xpath1)))
+    (into {}
+          (map (partial xml-elem->instrument-sensor-mapping doc) (select doc instruments-xpath2)))))
+
+(defn- get-child-instruments
+  "Returns the parsed child instruments from the instrument element.
+   The child instruments' mountedOn = the instrument element's id"
+  [doc instrument-elem]
+  (let [id (get-in instrument-elem [:attrs :id])
+        child-instruments-mapping (xml-elem->child-instruments-mapping doc)]
+    (remove nil?
+      (for [[k v] child-instruments-mapping]
+        (when (.contains k (str "-" id))
+          v)))))  
+
+;;(defn- get-ops-modes
+ ;; "Returns the operational modes"
+  ;;[])  
+
+(defn- xml-elem->instrument
+  [doc instrument-elem]
+  (let [child-instruments (get-child-instruments doc instrument-elem)]
+    (util/remove-nil-keys
+    (if (> (count child-instruments) 0)
+      {:ShortName (value-of instrument-elem iso/short-name-xpath)
+       :LongName (value-of instrument-elem iso/long-name-xpath)
+       :Characteristics (char-and-opsmode/parse-characteristics instrument-elem)
+       :Technique (without-default-value-of instrument-elem "gmi:type/gco:CharacterString")
+       :NumberOfInstruments (count child-instruments) 
+       :ComposedOf child-instruments
+       :OperationalModes (char-and-opsmode/parse-operationalmodes instrument-elem)}
+      {:ShortName (value-of instrument-elem iso/short-name-xpath)
+       :LongName (value-of instrument-elem iso/long-name-xpath)
+       :Characteristics (char-and-opsmode/parse-characteristics instrument-elem)
+       :Technique (without-default-value-of instrument-elem "gmi:type/gco:CharacterString")
+       :OperationalModes (char-and-opsmode/parse-operationalmodes instrument-elem)}))))
+
+(defn- xml-elem->instrument-mapping
```

Comment:
Done.  This function will find the all the instruments, excluding the child instruments(defined as having the mountedOn-id referring to any instrument ids), creating a map of its instrument id and the instrument record.
This will allow the platform to find all the instruments it's linking to.